libs/net/dal/Models/UserFilter.cs

```diff
@@ -15,6 +14,9 @@ public class UserFilter : PageFilter
     public Entities.UserStatus? Status { get; set; }
     public bool? IsSystemAccount { get; set; }
     public string[] Sort { get; set; } = Array.Empty<string>();
+    public string? Keyword { get; set; }
+    public IEnumerable<Entities.Role> Roles { get; set; } = Array.Empty<Entities.Role>();
```

Comment:
I don't think you should be filtering on role objects.  You should just be filtering on role names.